c/CMakeLists.txt

```diff
@@ -48,12 +48,23 @@ if(SUBSTRAIT_VALIDATOR_BUILD_TESTS)
         substrait-validator-c-test
         ${CMAKE_CURRENT_SOURCE_DIR}/tests/test.cc
     )
+
+    # For OSX, link in CoreFoundation, needed for some system symbols like
+    # _CFRelease, _CFStringGetBytes, …
```

Comment:
This was outside the intent of this PR, but after running into [this error](https://github.com/substrait-io/substrait-validator/actions/runs/11525385202/job/32087549504?pr=281) and attempting to reproduce locally, I found this option helped. I know very little about CMake, so any recommendations here welcome 🙇